src/core/core_utils.js

```diff
@@ -146,6 +146,36 @@ function getInheritableProperty({
   return values;
 }
 
+/**
+ * Get the parent dictionary to update when a property is set.
+ *
+ * @param {Dict} dict - Dictionary from where to start the traversal.
+ * @param {Ref} ref - The reference to the dictionary.
+ * @param {XRef} xref - The `XRef` instance.
+ */
+function getParentToUpdate(dict, ref, xref) {
+  const visited = new RefSet();
+  const firstDict = dict;
+  const result = { dict: null, ref: null };
+
+  while (dict instanceof Dict && !visited.has(ref)) {
+    visited.put(ref);
+    if (dict.has("T")) {
+      break;
+    }
+    ref = dict.getRaw("Parent");
+    if (!(ref instanceof Ref)) {
+      return result;
+    }
+    dict = xref.fetch(ref);
+  }
+  if (dict !== firstDict) {
```

Comment:
In a corrupt PDF, isn't it possible for the `dict`-fetching here to be *any* value (and not necessarily a dictionary)?

Let's assume that it's e.g. the number `5`, and in that case we'd break out of the loop just fine. However the `dict !== firstDict` check would be true and we'd thus incorrectly update the `result`, hence we'd run into trouble when trying to `clone` a non-dictionary in the annotation-code.

Unless I'm missing something, I do believe that we *also* need to check the type here as well.
```suggestion
    dict = xref.fetch(ref);
  }
  if (dict instanceof Dict && dict !== firstDict) {
```